Libcosim fails to compile with Visual Studio 2019

Visual Studio 2019 report warnings during compilation of libcosim:

`warning C5205: delete of an abstract class 'cosim::function' that has a non-virtual destructor results in undefined behavior`

This is reported for every class implementing the `function` interface and by default warnings will fail the build.

--- a/include/cosim/function/function.hpp
+++ b/include/cosim/function/function.hpp
@@ -169,6 +169,8 @@ public:
 
     /// Performs the function calculations.
     virtual void calculate() = 0;
+
+    virtual ~function() = default;
 };
 
 
@@ -203,6 +205,8 @@ public:
      */
     virtual std::unique_ptr<function> instantiate(
         const function_parameter_value_map& parameters) = 0;
+
+    virtual ~function_type() = default;
 };
 
 

